apps/src/templates/teacherDashboard/MultiSelectGroup.jsx

```diff
@@ -0,0 +1,98 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import _ from 'lodash';
+import i18n from '@cdo/locale';
+import {multiSelectOptionShape} from './shapes';
+import styles from './multi-select-group.module.scss';
+
+// NOTE: The `name` will show up in the DOM with an appended `[]`, so Rails
+// natively understands it as an array. Set `required` to `true` if you want
+// the user to have to select at least one of the options to proceed.
+// For an example of `values` format, see this component's storybook file.
+export default function MultiSelectGroup({
+  label,
+  name,
+  required,
+  options,
+  values,
+  setValues
+}) {
+  const inputName = `${name}[]`;
+
+  return (
+    <div className={styles.multiSelectGroup}>
+      <fieldset>
+        <label>{label}</label>
+        {options.map((option, index) => (
+          <MultiSelectButton
+            label={option.label}
+            name={inputName}
+            value={option.value}
+            key={option.value}
+            checked={values[option.value]}
+            // The child's `required` prop will be set to `false` if the
+            // Group's `required` prop is falsy. It will be set to `true` if
+            // the Group's `required` prop is truthy AND none of the options
+            // are `checked`, or `false` if at least one of the options is
+            // `checked`.
+            required={required ? !Object.values(values).some(v => !!v) : false}
+            onCheckedChange={checked => {
+              setValues({
+                ...values,
+                [option.value]: checked
+              });
+            }}
+          />
+        ))}
+      </fieldset>
+    </div>
+  );
+}
+
+function MultiSelectButton({
```

Comment:
Is it worth writing a test for this component? (it's pretty simple so maybe it isn't)

Reply:
I could set up some spies and just verify the expected functions are called with `onChange`/`onInvalid`, if that seems helpful? i wasn't sure if it was worth exporting this component at all

Reply:
It's probably fine without a test, given that it's just an input